internal/internal_activity.go

```diff
@@ -302,98 +302,131 @@ func isActivityContext(inType reflect.Type) bool {
 	return inType != nil && inType.Implements(contextElem)
 }
 
-func validateFunctionAndGetResults(f interface{}, values []reflect.Value, dataConverter converter.DataConverter) (*commonpb.Payloads, error) {
-	resultSize := len(values)
-
-	if resultSize < 1 || resultSize > 2 {
-		fnName, _ := getFunctionName(f)
-		return nil, fmt.Errorf(
-			"the function: %v signature returns %d results, it is expecting to return either error or (result, error)",
-			fnName, resultSize)
+func setActivityParametersIfNotExist(ctx Context) Context {
+	params := getActivityOptions(ctx)
+	var newParams ExecuteActivityOptions
+	if params != nil {
```

Comment:
Curious, what's the reason that you want to clone the old option?

Reply:
This is not my code. This remained unchanged from further down in the file. I would not have written the coed like this. Unfortunately git diff is confused with all of the changes in here.

Reply:
ok thanks